File: C2110I/MVC/30-01-2023/StudentManagement/Models/Student.cs
using Microsoft.Build.Framework;
using StudentManagement.Validations;
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;

namespace StudentManagement.Models
{
    public class Student
    {

        public int StudentID { get; set; }
        [DisplayName("Student's name")]
        public string StudentName { get; set; }        
        
        [EmailAddress(ErrorMessage ="Invalid email addresssssss")]
        public string Email { get; set;}

        [StringLength(200, MinimumLength =4, ErrorMessage ="Address must be 4 to 200 in length")]
        public string? Address { get; set; }
        [DataType(DataType.DateTime, ErrorMessage ="DOB must be Datetime")]
        [AgeGreaterThan(18, "Age must be >= 18")]
        public DateTime DOB { get; set; }

        //navigation property
        public ICollection<Enrollment> Enrollments { get; set; }
    }
}

File: TranQuangHung/UniversityApp/src/main/java/org/university/app/models/Student.java
package org.university.app.models;

public class Student {
    private int studentID;
    private String name;

    public Student(int studentID, String name) {
        this.studentID = studentID;
        this.name = name;
    }

    public int getStudentID() { return studentID; }
    public String getName() { return name; }
}
